### PR TITLE
Parially revert change that made some tags not able to be empty

### LINKED ISSUE #133: Allow empty tags in addon.xml
Some one decided that empty tags like website,forum,email are not allowed. That's just wrong and needs to be fixed as they are perfectly valid.

### PR DESCRIPTION
**Pull request by Razzeee**
Closes #133

### CODE REVIEW

File: kodi_addon_checker/xml_schema/gotham_metadata.xsd [hunk 1]
```diff
@@ -8,12 +8,12 @@
         <xs:element name="description" type="translatedString" minOccurs="0" maxOccurs="unbounded"/>
         <xs:element name="disclaimer" type="translatedString" minOccurs="0" maxOccurs="unbounded"/>
         <xs:element name="platform" type="platformList" minOccurs="0" maxOccurs="1"/>
-        <xs:element name="language" type="nonEmptyString"  minOccurs="0" maxOccurs="1"/>
+        <xs:element name="language" type="xs:string"  minOccurs="0" maxOccurs="1"/>
         <xs:element name="license" type="nonEmptyString"  minOccurs="0" maxOccurs="1" />
-        <xs:element name="forum" type="nonEmptyString" minOccurs="0" maxOccurs="1"/>
-        <xs:element name="website" type="nonEmptyString" minOccurs="0" maxOccurs="1"/>
+        <xs:element name="forum" type="xs:string" minOccurs="0" maxOccurs="1"/>
+        <xs:element name="website" type="xs:string" minOccurs="0" maxOccurs="1"/>
```
Comment (Rechi) [inline kodi_addon_checker/xml_schema/gotham_metadata.xsd]: Wiki doesn't say that website can be left blank.

Comment (Razzeee) [reply kodi_addon_checker/xml_schema/gotham_metadata.xsd]: Should be resolved

File: kodi_addon_checker/xml_schema/gotham_metadata.xsd [hunk 2]
```diff
@@ -8,14 +8,14 @@
         <xs:element name="description" type="translatedString" minOccurs="0" maxOccurs="unbounded"/>
         <xs:element name="disclaimer" type="translatedString" minOccurs="0" maxOccurs="unbounded"/>
         <xs:element name="platform" type="platformList" minOccurs="0" maxOccurs="1"/>
-        <xs:element name="language" type="nonEmptyString"  minOccurs="0" maxOccurs="1"/>
+        <xs:element name="language" type="xs:string"  minOccurs="0" maxOccurs="1"/>
```
Comment (enen92) [inline kodi_addon_checker/xml_schema/gotham_metadata.xsd]: I'd leave language as a `nonEmptyString `. It is often a source of confusion between addon submitters and if included we should force users to define it. We can always edit the wiki

Comment (Razzeee) [reply kodi_addon_checker/xml_schema/gotham_metadata.xsd]: Can you please go ahead do that and edit the wiki? I will change the PR then.

**Comment by Rechi** [comment]
I actually meant the following while talking about news
https://github.com/xbmc/addon-check/blob/9d9b8d89159cfa86b33a06962a0daf8cd81afad1/kodi_addon_checker/xml_schema/gotham_metadata.xsd#L35

**Comment by Razzeee** [comment]
I did understand that. So you want me to make it mandatory to fill it, but not to cap it?

File: kodi_addon_checker/xml_schema/gotham_metadata.xsd (hunk 2 above)
Comment (Rechi) [reply kodi_addon_checker/xml_schema/gotham_metadata.xsd]: So what's the difference to allow empty forum, email and news, but not language.
Either allow all to be empty, which in my opinion gives no additional info and bloats the addon.xml, or require all to be non empty or not added.

Comment (enen92) [reply kodi_addon_checker/xml_schema/gotham_metadata.xsd]: done

File: kodi_addon_checker/xml_schema/gotham_metadata.xsd [hunk 3]
```diff
@@ -8,20 +8,20 @@
...
-        <xs:element name="source" type="nonEmptyString" minOccurs="0" maxOccurs="1"/>
-        <xs:element name="email" type="nonEmptyString" minOccurs="0" maxOccurs="1"/>
-        <xs:element name="broken" type="nonEmptyString" minOccurs="0" maxOccurs="1"/>
+        <xs:element name="language" type="nonEmptyStringCapped"  minOccurs="0" maxOccurs="1"/>
+        <xs:element name="license" type="nonEmptyStringCapped"  minOccurs="0" maxOccurs="1" />
+        <xs:element name="forum" type="xs:string" minOccurs="0" maxOccurs="1"/>
+        <xs:element name="website" type="nonEmptyStringCapped" minOccurs="0" maxOccurs="1"/>
+        <xs:element name="source" type="nonEmptyStringCapped" minOccurs="0" maxOccurs="1"/>
+        <xs:element name="email" type="xs:string" minOccurs="0" maxOccurs="1"/>
+        <xs:element name="broken" type="nonEmptyStringCapped" minOccurs="0" maxOccurs="1"/>
```
Comment (Rechi) [inline kodi_addon_checker/xml_schema/gotham_metadata.xsd]: no max length mentioned for any of those elements in the wiki, they can be as long as wanted

File: kodi_addon_checker/xml_schema/gotham_metadata.xsd (hunk 2 above)
Comment (MartijnKaijser) [reply kodi_addon_checker/xml_schema/gotham_metadata.xsd]: language must be allowed empty.

Comment (enen92) [reply kodi_addon_checker/xml_schema/gotham_metadata.xsd]: @Rechi I think language (unlike the others) is the only used in the Kodi GUI...likely ignored if empty though. I agree with you that empty tags only bloat the addon.xml but suddenly blocking user PRs is not a good alternative either. In the end, if Kodi does not complain it is up to the developer to have such a verbose manifest. We can warn during reviews if we spot such stuff

File: kodi_addon_checker/xml_schema/gotham_metadata.xsd (hunk 3 above)
Comment (Razzeee) [reply kodi_addon_checker/xml_schema/gotham_metadata.xsd]: The initial trigger for this was https://github.com/xbmc/addon-check/pull/118 which was a limit for the news section. We need to change the wiki and at that point we can also think of limiting the other fields. But it's likely that we want shorter limits.

Until then I'm going to not cap them.

File: kodi_addon_checker/xml_schema/gotham_metadata.xsd [hunk 4]
```diff
@@ -32,12 +32,18 @@
   <xs:simpleType name="nonEmptyString">
     <xs:restriction base="xs:string">
       <xs:pattern value="\s*([^\s]\s*)+"/>
-      <xs:maxLength value="1500"/>  
     </xs:restriction>
   </xs:simpleType>
-  <xs:complexType name="translatedString">
+  <xs:complexType name="nonEmptyStringCapped">
```
Comment (Rechi) [inline kodi_addon_checker/xml_schema/gotham_metadata.xsd]: no need for nonEmptyString and nonEmptyStringCapped, they are both the same

File: kodi_addon_checker/xml_schema/gotham_metadata.xsd [hunk 5]
```diff
@@ -10,12 +10,12 @@
...
         <xs:element name="license" type="nonEmptyString"  minOccurs="0" maxOccurs="1" />
-        <xs:element name="forum" type="nonEmptyString" minOccurs="0" maxOccurs="1"/>
+        <xs:element name="forum" type="xs:string" minOccurs="0" maxOccurs="1"/>
         <xs:element name="website" type="nonEmptyString" minOccurs="0" maxOccurs="1"/>
         <xs:element name="source" type="nonEmptyString" minOccurs="0" maxOccurs="1"/>
-        <xs:element name="email" type="nonEmptyString" minOccurs="0" maxOccurs="1"/>
+        <xs:element name="email" type="xs:string" minOccurs="0" maxOccurs="1"/>
         <xs:element name="broken" type="nonEmptyString" minOccurs="0" maxOccurs="1"/>
-        <xs:element name="news" type="nonEmptyString" minOccurs="0" maxOccurs="1"/>
+        <xs:element name="news" type="nonEmptyStringCapped" minOccurs="0" maxOccurs="1"/>
```
Comment (Rechi) [inline kodi_addon_checker/xml_schema/gotham_metadata.xsd]: wiki doesn't mention a limit

**Comment by Razzeee** [comment]
Okay, another update. Can we agree, to do it like this and change the wiki?

**Comment by Rechi** [comment]
If we change the wiki, I still not get why we would then require non empty strings for license, source and website, but not for email, forum, news and language.

I thought this whole PR is to keep the wiki as is, because otherwise the wiki could be changed for all elements and this PR wouldn't be needed.

**Comment by MartijnKaijser** [comment]
What is current not according to the wiki?

**Comment by Rechi** [comment]
With the current state of this PR, max length for
- description
- disclaimer
- news
- summary

**Comment by Razzeee** [comment]
Current state of PR only max lengths:
* news

But I think the non empty stuff (no empty tags) for these are also not documented:
* license
* website
* source
* broken
* news (again)
* addon name
* icon
* fanart
* screenshot
* clearlogo
* banner

**Comment by Razzeee** [comment]
I would like to merge this as is any more comments?

**Comment by Rechi** [comment]
As already mentioned a few times in this PR, the wiki doesn't mention a char limit for news.

**Comment by Razzeee** [comment]
Yes, thats what I'm going to change as soon as we merge this.

**Pull request merged**